### PR TITLE
🎮 Support terrains without otc file

### LINKED ISSUE #1841: Compatibility with terrains with missing otc file
Try to load a terrain with missing otc file:
```
[RoR|Terrain] Error reading main *.otc file, message: 'FileNotFoundException: Cannot locate resource Houston2013.otc in resource group General. in ResourceGroupManager::openResource at /home/babis/Downloads/ror-dependencies/Source/ogre/OgreMain/src/OgreResourceGroupManager.cpp (line 698)', type: <Ogre::Exception>.
```
Example terrain: http://www.mediafire.com/file/nn36bdj9bo99khv/Traxxas_Stadium_Freestle_4.zip/file

It would be nice if this will work again, like older RoR versions (pre 0.4.7)

### PR DESCRIPTION
**Pull request by ulteq**
We skip the entire terrain initialization if the `otc` file is missing.

All credits go to @AnotherFoxGuy.

Closes: https://github.com/RigsOfRods/rigs-of-rods/pull/1849
Closes #1841

### CODE REVIEW

File: source/main/resources/terrn2_fileformat/Terrn2Fileformat.cpp [hunk 1]
```diff
@@ -72,6 +65,14 @@ bool Terrn2Parser::LoadTerrn2(Terrn2Def& def, Ogre::DataStreamPtr &ds)
 
     def.start_position       = StringConverter::parseVector3(file.GetStringEx("StartPosition", "General"), Vector3(512.0f, 0.0f, 512.0f));
 
+    Ogre::FileSystemArchiveFactory factory;
+    Ogre::Archive* fs_archive = factory.createInstance(def.ter_conf_filename, /*readOnly*/true);
+    if (!fs_archive->exists(def.ter_conf_filename))
+    {
+        def.ter_conf_filename = "";
```
Comment (only-a-ptr) [inline source/main/resources/terrn2_fileformat/Terrn2Fileformat.cpp]: This does deserve a warning in RoR.log

File: source/main/terrain/TerrainManager.cpp [hunk 2]
```diff
@@ -401,13 +408,15 @@ void TerrainManager::initWater()
 
         m_water = std::unique_ptr<IWater>(m_hydrax_water);
 
-        //Apply depth technique to the terrain
-        TerrainGroup::TerrainIterator ti = m_geometry_manager->getTerrainGroup()->getTerrainIterator();
-        while (ti.hasMoreElements())
+        if (m_geometry_manager->getTerrainGroup())
```
Comment (only-a-ptr) [inline source/main/terrain/TerrainManager.cpp]: This needs a `IsGeometryEnabled()` helper to clarify what's going on.

File: source/main/terrain/TerrainGeometryManager.cpp [hunk 3]
```diff
@@ -401,6 +402,9 @@ bool TerrainGeometryManager::InitTerrain(std::string otc_filename)
 
 void TerrainGeometryManager::updateLightMap()
 {
+    if (m_ogre_terrain_group == nullptr)
```
Comment (only-a-ptr) [inline source/main/terrain/TerrainGeometryManager.cpp]: This needs a `IsGeometryEnabled()` helper to clarify what's going on.

File: source/main/terrain/TerrainGeometryManager.cpp [hunk 4]
```diff
@@ -419,6 +423,9 @@ void TerrainGeometryManager::updateLightMap()
 
 void TerrainGeometryManager::UpdateMainLightPosition()
 {
+    if (m_ogre_terrain_group == nullptr || TerrainGlobalOptions::getSingletonPtr() == nullptr)
```
Comment (only-a-ptr) [inline source/main/terrain/TerrainGeometryManager.cpp]: ditto

**Comment by ulteq** [comment]
I think we can't do it like that, we found issues with maps like Badlands Rally ... currently investigating.

**Comment by AnotherFoxGuy** [comment]
What was wrong with #1849?

**Comment by ulteq** [comment]
I didn't understand why we have to change so much only to fix this little problem.

**Comment by ulteq** [comment]
This is crap.

**Pull request closed without merging**